### PR TITLE
Clarify LXD install and improve link/title stuff

### LINKED ISSUE #2544: Ensure references to LXD install methods/versions are sane
For instance, charms-storage.md includes (with a broken link):

"Note: To get an LXD version (at least 2.16) on either Ubuntu 14.04 LTS (Trusty) or Ubuntu 16.04 LTS (Xenial) that has the 'lxd' storage provider feature the LXD:stable PPA will be required."

Suggestion: make use of install details included within clouds-LXD.md.

### PR DESCRIPTION
**Pull request by pmatulis**
resolves #2544

### CODE REVIEW

File: src/en/clouds.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ There are three special types of clouds: MAAS, LXD and Manual.
   developing your own Juju charms - it is incredibly fast! 
   [LXD is a container hypervisor][LXD-site] that runs on any Linux host, providing 
   the ability to spin up containers on the host machine. For more details on
-  using LXD, please see the [LXD documentation][juju-lxd].
+  using LXD, please see the [Using LXD with Juju][clouds-lxd].
```
Comment (degville) [inline src/en/clouds.md]: needs *page* or *documentation* before the end stop.

File: src/en/charms-storage.md [hunk 2]
```diff
@@ -466,9 +466,9 @@ provider currently supports a single pool configuration attribute:
 ### LXD (lxd)
 
 !!! Note:
-    To get an LXD version (at least 2.16) on either Ubuntu 14.04 LTS (Trusty)
-    or Ubuntu 16.04 LTS (Xenial) that has the 'lxd' storage provider feature
-    the [LXD:stable PPA][ppa-lxd] will be required.
+    Ubuntu 14.04 LTS (Trusty) and Ubuntu 16.04 LTS (Xenial) do not ship with a
+    version of LXD (at least 2.16) that has the 'lxd' storage provider feature.
```
Comment (degville) [inline src/en/charms-storage.md]: I found the version number slightly confusing on first read, like it's the version found in those old distros. I'd move *(at least 2.16)* after the explanation that you need LXD with a storage provider and take it out of brackets, such as *,available in LXD 2.16 and later.*

File: src/en/introducing-2.md [hunk 3]
```diff
@@ -35,8 +35,7 @@ cloud you only need one controller, but can drive several models.
  - **Providers:** As well as updates to the Azure provider and 
 the new Rackspace provider, Juju now makes extensive use of LXD. This makes 'local' 
 models leaner and faster and also brings benefits to placing containers within 
-models. You can find out more about setting up 
-[LXD for Juju here][lxd].
+models. You can find out more about [Using LXD with Juju][clouds-lxd].
```
Comment (degville) [inline src/en/introducing-2.md]: The literal meaning seems broken without the link, eg. You can find out more about[Using LXD with Juju (where?).

File: src/en/clouds.md (hunk 1 above)
Comment (pmatulis) [reply src/en/clouds.md]: :astonished:

File: src/en/charms-storage.md (hunk 2 above)
Comment (pmatulis) [reply src/en/charms-storage.md]: :+1:

File: src/en/introducing-2.md (hunk 3 above)
Comment (pmatulis) [reply src/en/introducing-2.md]: :astonished:

**Pull request merged**